src/main/kotlin/uk/gov/justice/digital/hmpps/supportadditionalneedsapi/service/ReviewScheduleService.kt

```diff
@@ -97,24 +106,62 @@ class ReviewScheduleService(
+    val updatedDeadlineDate = getUpdatedDeadlineDate(current, prisonNumber, startDate, proposedDeadline)
 
-    if (current != desired) {
-      existing.deadlineDate = desired
+    // If it has changed then update the date and publish messages otherwise just log that it didn't change
+    if (current != updatedDeadlineDate) {
+      existing.deadlineDate = updatedDeadlineDate
       reviewScheduleRepository.save(existing)
       eventPublisher.createAndPublishReviewScheduleEvent(prisonNumber)
-      log.info("Review schedule deadline date updated to $desired for $prisonNumber, days to add was configured as $reviewConfig.reviewDeadlineDaysToAdd days")
+      log.info("Review schedule deadline date updated to $updatedDeadlineDate for $prisonNumber, days to add was configured as ${reviewConfig.reviewDeadlineDaysToAdd} days")
     } else {
-      log.info("Review date was unchanged for $prisonNumber the current deadline date was $current and the proposed was $proposedDeadline, days to add was configured as $reviewConfig.reviewDeadlineDaysToAdd days")
+      log.info("Review date was unchanged for $prisonNumber the current deadline date was $current and the proposed was $proposedDeadline, days to add was configured as ${reviewConfig.reviewDeadlineDaysToAdd} days")
+    }
+  }
+
+  private fun getUpdatedDeadlineDate(
+    current: LocalDate,
+    prisonNumber: String,
+    startDate: LocalDate,
+    proposedDeadline: LocalDate,
+  ): LocalDate {
+    var updatedDeadlineDate = minOf(current, proposedDeadline)
+    if (current != updatedDeadlineDate) {
+      if (reviewConfig.additionalReviewDateLogic) { // TODO remove this IF guard and config when we are happy with the logic
+        // At this point in the code the person has an existing review and has been put on another course
+        // If the start date of the new course is earlier than the date that the plan was created then we should
+        // not update the review date and keep it as is.
+        if (current != updatedDeadlineDate) {
+          val planCreationInstant = elspPlanRepository.findByPrisonNumber(prisonNumber)
+            ?.createdAt
+          planCreationInstant
+            ?.atZone(ZoneId.systemDefault())
+            ?.toLocalDate()
+            ?.takeIf { planCreationDate -> startDate < planCreationDate }
+            ?.let { updatedDeadlineDate = current } // short circuit so that the deadline date isn't changed.
+        }
+      }
+
+      // or if the most recent review was after the start date
+      if (current != updatedDeadlineDate) {
+        val reviewCreatedInstant = elspReviewRepository.findFirstByPrisonNumberOrderByUpdatedAtDesc(prisonNumber)
+          ?.createdAt
+        reviewCreatedInstant
+          ?.atZone(ZoneId.systemDefault())
+          ?.toLocalDate()
+          ?.takeIf { reviewCreatedDate -> reviewCreatedDate < current }
+          ?.let { updatedDeadlineDate = current } // short circuit so that the deadline date isn't changed.
+      }
     }
```

Comment:
This section is the change - 

if there is a plan then see if this new education start date is before the plan created date and if so stop here. 

Then if the date has changed do the same with the most recent review.